File: program_data/github_java_sort_function/7/35.java
package com.algorithmhelper.generalalgorithms.sorting;

public class ShellSort {

    
    public static void sort(Comparable[] arr) {
        if (arr == null)
            throw new IllegalArgumentException("sort with null arr");

        int n = arr.length;
        int h = 1;
        while (h < n/3)
            h = 3*h + 1;

        while (h >= 1) {
            for (int i = h; i < n; i++) {
                for (int j = i; j >= h && less(arr[j], arr[j-h]); j -= h)
                    swap(arr, j, j-h);
            }
            h /= 3;
        }
    }

    
    private static boolean less(Comparable x, Comparable y) {
        return x.compareTo(y) < 0;
    }

    
    private static void swap(Comparable[] arr, int i, int j) {
        Comparable temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}

File: program_data/PKU_raw/88/35.c
int main()
{
	char s[31],temp[31];
	int j=0,c=0;
	gets(s);
	while(s[j]!='\0')
	{
		if(s[j]>='0'&&s[j]<='9')
		temp[c++]=s[j];
		else 
		{
			temp[c]='\0';
			c=0;
			printf("%s\n",temp);
		}
		j++;
	}
	temp[c]='\0';
	printf("%s",temp);
	return 0;
}
		
		
	
